File: ros_rp2w/src/rp2w_pub.py
#!/usr/bin/env python
# license removed for brevity
import rospy
from ros_rp2w.msg import AdvancedCommand

if __name__ == '__main__':
    pub = rospy.Publisher("rp2w/advanced_command", AdvancedCommand, queue_size=1)
    rospy.init_node('rp2w_publisher', anonymous=True)
    rate = rospy.Rate(10) # 10hz
    # while not rospy.is_shutdown():
    while pub.get_num_connections() == 0:
        rate.sleep()
    msg = AdvancedCommand()
    msg.distanceCommand = True
    msg.distance = 1
    msg.thetaCommand = False
    msg.theta = 0
    pub.publish(msg)
    print msg

File: ros_rp2w/src/rp2w_pub.cpp
#include "ros/ros.h"
#include "ros_rp2w/AdvancedCommand.h"

using namespace std;


int main(int argc, char **argv)
{
  ros::init(argc, argv, "rp2w_pub");
  ros::NodeHandle n;
  ros::Publisher pub = n.advertise<ros_rp2w::AdvancedCommand>("rp2w/advanced_command", 1);
  // ros::Rate loop_rate(10);

  ros_rp2w::AdvancedCommand msg;
  bool distanceCommand = true;
  int distance = 1;
  bool thetaCommand = false;
  int theta = 0;
  msg.distanceCommand = distanceCommand;
  msg.distance = distance;
  msg.thetaCommand = thetaCommand;
  msg.theta = theta;
  pub.publish(msg);

  // cout << "distanceCommand: " << distanceCommand
  //      << "\tdistance: " << distance
  //      << "\tthetaCommand: " << thetaCommand
  //      << "\ttheta: " << theta << endl;

  // while (ros::ok())
  // {
  //   /**
  //    * This is a message object. You stuff it with data, and then publish it.
  //    */
  //   std_msgs::String msg;

  //   std::stringstream ss;
  //   ss << "hello world " << count;
  //   msg.data = ss.str();

  //   ROS_INFO("%s", msg.data.c_str());
     
  //   chatter_pub.publish(msg);

  //   ros::spinOnce();

  //   loop_rate.sleep();
  //   ++count;
  // }


  return 0;
}
